hivemind/utils/networking.py

```diff
@@ -1,12 +1,16 @@
 import socket
 from contextlib import closing
 from ipaddress import ip_address
-from typing import Optional, Sequence
+from typing import Optional, Sequence, Tuple
 
 from multiaddr import Multiaddr
 
 Hostname, Port = str, int  # flavour types
-Endpoint = str  # e.g. 1.2.3.4:1337 or [2a21:6с8:b192:2105]:8888, https://networkengineering.stackexchange.com/a/9435
+Endpoint = (
+    Tuple[  # e.g. 1.2.3.4:1337 or [2a21:6с8:b192:2105]:8888, https://networkengineering.stackexchange.com/a/9435
+        str, Tuple[str, ...]
+    ]
+)
```

Comment:
Ideally, we should always use `PeerID` instead of `Endpoint`, so libp2p will choose a route to the target peer by itself (as I explained during our call).

If we postpone this change to the next PR, we still don't such an `Endpoint` tuple since it duplicates the existing `hivemind.PeerInfo` structure (but worse since it doesn't have the named fields).

Also, the comment here became irrelevant.